app/src/main/java/com/github/wnder/LoginActivity.java

```diff
@@ -33,6 +33,8 @@ protected void onCreate(Bundle savedInstanceState) {
         signInButton = findViewById(R.id.sign_in_button);
         signInButton.setVisibility(View.INVISIBLE);  // Hide the button
         signInButton.setOnClickListener(this);
+        findViewById(R.id.guest_button).setOnClickListener(this);
```

Comment:
Shouldn't we provide onClickListeners instead of an Activity ?

Reply:
I admit this looks a bit strange but it allows to use a single onClick method in the activity to handle any button click with a simple switch statement so I think it has its advantages.